driver/driver.go

```diff
@@ -38,7 +40,17 @@ func FromSource(source models.Source) (Driver, error) {
 		var creds *credentials.Credentials
 
 		if source.AccessKeyID == "" && source.SecretAccessKey == "" {
-			creds = credentials.AnonymousCredentials
+			// Try with instance profile
+			creds = credentials.NewCredentials(
+				&ec2rolecreds.EC2RoleProvider{
+					Client: ec2metadata.New(session.New()),
+				})
+			cred_details, _ := creds.Get()
+
+			// If unsuccessful fall back to anonymous
+			if cred_details == (credentials.Value{}) {
```

Comment:
Does it make sense to see if the ignored error on the line above isn't `nil` instead?

Reply:
This was lifted and shifted from the S3 resource version. Now that I think about it more, the `Get` function returns a concrete `Value` struct, not a pointer, so this will either contain some credentials, or be a blank one. In this use case, we aren't interested in whether it errored or succeeded, but found no credentials - in either case we want to fall back to using the Anonymous creds.

I guess we could write it as:
```go
cred_details, err := creds.Get()
    // If unsuccessful fall back to anonymous
    if err != nil || cred_details == (credentials.Value{}) {
        // ...
```
but that wouldn't behave any differently.

Thinking about it further, and having read the code, it's probably clear to check the error as you suggest.  I'll update it.

Reply:
Updated...